pkg/server/server.go

```diff
@@ -579,17 +579,26 @@ func (s *Server) HTTPHandler() http.Handler {
 		})
 	}
 
+	multiClusterHandler := func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			cluster := serverutils.GetCluster(r)
+			helmHandlers.ApiServerHost = s.K8sProxyConfigs[cluster].Endpoint.String()
+			helmHandlers.Transport = s.K8sClients[cluster].Transport
```

Comment:
No I don't think this is thread safe since we're changing the state of the server with each request. We probably need a solution that refactors the helm handler package to include mulitcluster config that sets up the handlers for each managed cluster at startup. Requests then can be directed to a pre-configured handler for the appropriate cluster without changing any state on the server.